### PR TITLE
Remove companion object KEY in all Screens

### PR DESCRIPTION
**Pull request by sanchita**

### CODE REVIEW

File: app/src/androidTest/java/org/simple/clinic/editpatient/PatientEditScreenUiTest.kt [hunk 1]
```diff
@@ -48,7 +48,7 @@ class PatientEditScreenUiTest {
     patientRepository.save(listOf(patientProfile)).blockingAwait()
 
     activityTestRule.runOnUiThread {
-      activityTestRule.activity.screenRouter.clearHistoryAndPush(PatientEditScreen.KEY(patientProfile.patient.uuid), RouterDirection.FORWARD)
+      activityTestRule.activity.screenRouter.clearHistoryAndPush(PatientEditScreen.(::PatientEditScreenKey)(patientProfile.patient.uuid), RouterDirection.FORWARD)
```
Comment (saket) [inline app/src/androidTest/java/org/simple/clinic/editpatient/PatientEditScreenUiTest.kt]: This is such a weird syntax

Comment (saket) [reply app/src/androidTest/java/org/simple/clinic/editpatient/PatientEditScreenUiTest.kt]: fixparrot

File: app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt [hunk 2]
```diff
@@ -109,7 +105,8 @@ class NewMedicalHistoryScreen(context: Context, attrs: AttributeSet) : RelativeL
           .map { SaveMedicalHistoryClicked() }
 
   fun openPatientSummaryScreen(patientUuid: UUID) {
-    screenRouter.push(PatientSummaryScreen.KEY(patientUuid, PatientSummaryCaller.NEW_PATIENT, Instant.now(clock)))
+    Instant.now(clock)
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt]: @vinaysshenoy looks like I've caught sanchita's intentional mistake to catch me if I skim her PR

File: app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt [hunk 3]
```diff
@@ -105,11 +101,12 @@ class PatientSearchResultsScreen(context: Context, attrs: AttributeSet) : Relati
   }
 
   fun openPatientSummaryScreen(patientUuid: UUID) {
-    screenRouter.push(PatientSummaryScreen.KEY(patientUuid, PatientSummaryCaller.SEARCH, Instant.now(clock)))
+    Instant.now(clock)
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt]: Aand another intentional mistake

File: app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt [hunk 4]
```diff
@@ -204,16 +204,16 @@ class TheActivityControllerTest {
   @Suppress("Unused")
   private fun `params for local user initial screen key`(): Array<Array<Any>> {
     return arrayOf(
-        arrayOf<Any>(true, NOT_LOGGED_IN, RegistrationPhoneScreen.KEY::class.java),
-        arrayOf<Any>(true, OTP_REQUESTED, HomeScreen.KEY::class.java),
-        arrayOf<Any>(true, RESET_PIN_REQUESTED, HomeScreen.KEY::class.java),
+        arrayOf<Any>(true, NOT_LOGGED_IN, RegistrationPhoneScreenKey()::class.java),
```
Comment (saket) [inline app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt]: You don't need to instantiate an object to get a class. `RegistrationPhoneScreenKey::class.java` should also work.

File: app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt [hunk 5]
```diff
@@ -232,8 +232,8 @@ class TheActivityControllerTest {
   @Suppress("Unused")
   private fun `params for missing user initial screen key`(): Array<Array<Any>> {
     return arrayOf(
-        arrayOf<Any>(true, RegistrationPhoneScreen.KEY::class.java),
-        arrayOf<Any>(false, OnboardingScreen.KEY::class.java)
+        arrayOf<Any>(true, RegistrationPhoneScreenKey()::class.java),
+        arrayOf<Any>(false, OnboardingScreenKey()::class.java)
```
Comment (saket) [inline app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt]: Same question here

File: app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt (hunk 4 above)
Comment (sanchita) [reply app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt]: Refactoring error

File: app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt (hunk 5 above)
Comment (sanchita) [reply app/src/test/java/org/simple/clinic/activity/TheActivityControllerTest.kt]: Refactoring error

File: app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt (hunk 3 above)
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt]: You passed!

**Comment by saket** [comment]
@vinaysshenoy because this PR touches **all** screen navigations in the app, could you also take a look at this? An additional pair of eyes would be nice.

**Pull request merged**